### ISSUE TITLE
[Feature request] Add Odysee videos to channels

### ISSUE THREAD

**Issue by Myzel394** [iv-org/invidious#3022]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Many YouTube channels mirror or extend their videos onto Odysee (see examples below), because they either want to provide a more privacy-friendly method to access their videos or they have to upload their videos onto the LBRY-blockchain because YouTube censors them. 

My solution would be that we should add an option to Invidious to allow users automatically being redirected to the corresponding Odysee video when clicking on a YouTube video, instead of viewing the video through the Indivious instance.

The only problem I see is how do we find out if the YouTube channel has an Odysee account? I'd propose that we check that using YouTube's banner links (no idea what they're called, [I mean those links you can see in the bottom right corner on a YouTube banner like on this channel](https://www.youtube.com/c/MentalOutlaw))

##### Examples
* The Linux Experiment: [Youtube](https://www.youtube.com/channel/UC5UAwBUum7CPN5buc-_N1Fw), [Odysee](https://odysee.com/@TheLinuxExperiment:e)
* Fireship: [Youtube](https://www.youtube.com/c/Fireship), [Odysee](https://odysee.com/@fireship:6)
* Mental Outlaw: [Youtube](https://www.youtube.com/c/MentalOutlaw), [Odysee](https://odysee.com/@AlphaNerd:8)

**Describe alternatives you've considered**
We could also add the option to only redirect to Odysee if the content is not available on YouTube.

**Additional context**
-

**Comment by IronMaltese** [comment]
I don't think that's an official hated one channel, he makes videos about how much he hates YouTube, but idk why he hasn't uploaded his stuff on odysee

**Comment by TheFrenchGhosty** [comment]
Our of scope. There's also no way to get the corresponding channel.

**Comment by Myzel394** [comment]
> Out of scope

Why out of scope? It's related to Indivious.

> There's also no way to get the corresponding channel.

I already suggested a method to get the correct channel. Please read my issue:
"I'd propose that we check that using YouTube's banner links"

**Comment by TheFrenchGhosty** [comment]
>Why out of scope? It's related to Indivious.

Invidious is a YouTube frontend, YouTube doesn't have this feature, so we won't have this feature.

>I already suggested a method to get the correct channel. Please read my issue:
>"I'd propose that we check that using YouTube's banner links"

I did read it, it isn't doable since it's not consistent.

**Comment by Myzel394** [comment]
> Invidious is a YouTube frontend, YouTube doesn't have this feature, so we won't have this feature.

Youtube doesn't have download functionality, audio mode or reddit comments either. Since Indivious is focused on privacy, it would be a good fit to redirect users if they want to

> I did read it, it isn't doable since it's not consistent.

We could take a look at how the extensions implemented it

> Note: browser addons are free to implement it though, if they REALLY want to... though, good luck.

[Just found out about this extension](https://addons.mozilla.org/en-US/firefox/addon/watch-on-odysee/), which, after a first glance, seems to work pretty well.

Btw, please stop using these sarcastic comments, that's not healthy for the community.

Seems like you generally dislike Odysee, which is totally okay since you don't have to use it. Still, I think this should be implemented with the ability to turn it off, in case you don't want to be redirected. Maybe someone else should take a look at this. @SamantazFox you seem to be very active, what do you think?

**Comment by TheFrenchGhosty** [comment]
>Youtube doesn't have download functionality, audio mode or reddit comments either. Since Indivious is focused on privacy, it would be a good fit to redirect users if they want to

The thing is that those features are "bonus" that are easy to implement and actually useful (though: reddit comments is legacy, it's kept because it works - and in the case of the download functionality, the plan is to cut most of it #1563 ). Invidious is focused on privacy, yes, but it's focused on staying close to what YouTube *should be*, we try to implement the least amount of features that can be easily done by plugins that already exist (Sponsorblock for example). Invidious is already too massive for us to implement more "fluff", and my personal opinion is that we should actually look into making it more minimal.

>[Just found out about this extension](https://addons.mozilla.org/en-US/firefox/addon/watch-on-odysee/), which, after a first glance, seems to work pretty well.

It is using a dedicated API made by the LBRY developers, that's good to know, I wasn't aware that it was a thing and just assumed it couldn't be possible, sorry about that.
This however confirms what I was saying previously: Invidious support should be implemented to this add-on, since it should be trivial to do.

>Btw, please stop using these sarcastic comments, that's not healthy for the community.

I wasn't trying to be sarcastic, I was trying to be quick/direct when replying because we often get this kind of "fluff" (I can't think of a better way to call them) feature requests, if it sounds like sarcasm, it wasn't on purpose. Sorry.

>Seems like you generally dislike Odysee, which is totally okay since you don't have to use it.

I mean, I dislike Odysee, yes, but I would have said the same thing if it was another platform. We don't implement features that can easily exist as browser add-on.

>Still, I think this should be implemented with the ability to turn it off, in case you don't want to be redirected.

Again, a browser add-on should handle it.

( @SamantazFox will surely agree with me, since it's how we've been handling feature requests)

**Comment by SamantazFox** [comment]
> @SamantazFox you seem to be very active, what do you think?

Well, I'm more nuanced on the subject.

On one side, I'd love to see many of those features to be implemented as some sort of "plug-ins", and make invidious a powerful platform for many different people in the community, and allow them to really tailor the experience to what they want.

But on the other side, I'm realistic. There are multiple reasons why we can't implement that:

* The biggest is that I'm currently the only (major) developer of Invidious, and I really don't have the time to implement those features. Keeping invidious stable and functional is already a lot of work (we're already missing so much stuff, and I can't keep up). More features = more code to test, debug, and keep functional.

* All those features should be done on the client side, with JS, otherwise the performance impact for most of our users would be too big (increased page load time, due to extra API requests). And I'm pretty bad at JS. And testing that invidious works on _so many_ devices is *painful*.

* The current code is not very flexible, meaning that this kind of features have to be deeply implanted in different parts of the software, and can't just be added as simple plug-ins.


So to sum up: we're counting on the community to develop (or at least adapt existing) tools in order to enhance Invidious. The current code is not flexible enough to allow the implementation of such feature easily (though, I'm working on cleaning that up), and the team is lacking developers.
